examples/java/src/main/java/co/acme/parse/Raml10Parsing.java

```diff
@@ -42,4 +42,28 @@ public static void parseString() throws InterruptedException, ExecutionException
     String output = Raml10.generateString(doc).get();
     System.out.println("Generated Raml10 string:\n" + output);
   }
+
+  // Example of parsing RAML 1.0 string with location param
+  public static void parseStringWithLocation() throws InterruptedException, ExecutionException {
+    String inp = "#%RAML 1.0\n" +
+                 "title: API with Types\n" +
+                 "/users/{id}:\n" +
+                 "  get:\n" +
+                 "    responses:\n" +
+                 "      200:\n" +
+                 "        body:\n" +
+                 "          application/json:\n" +
+                 "            type: !include cat-schema.json";
+    String location = "file://../api-specs/includes/api.raml";
```

Comment:
`api.raml` does not exist. even if it did, this wouldn't be the file that is being parsed so it doesn't "feel right". This should be a "base location" or "base path" rather that a "location" since there is effectively no location with a string. In thse case: `file://../api-specs/includes/` instead of `file://../api-specs/includes/api.raml`

Reply:
Should I rename all "location" arguments to "baseLocation"?

This will still be `file://../api-specs/includes/api.raml` because initially this tasks purpose was to give ramldt2jsonscheme a way to reimplement `setBasePath`. My mistake was that I mentioned "base path" in webapi-parser task when I was actually going to add a way to pass "url" [1] param to AMF parsers when parsing a string. But since we already have a param called "urlOrContent", I called the new param "location".

> This should be a "base location" or "base path" rather that a "location" since there is effectively no location with a string

Right, and this param (currently called "location") sets model location when it's parsed from a string.

[1] https://github.com/aml-org/amf-core/blob/2f34c462260bfe4d34ed33c44dd8e43a5359a776/shared/src/main/scala/amf/client/parse/Parser.scala#L63

Reply:
I see. In that case, let's call it `baseUrl` instead. Also, here and in the other examples, I would set it to `file://base/url/` instead of `file://base/url/api.raml` to avoid confusion. I say "confusion" because it may lead people to think that we are parsing `file://base/url/api.raml` when we are not.

Reply:
It can't be a folder path. I've tested folder path as location argument in js example and it failed to resolve references.
It must end with a file name, at least a fake one. In this case `file://base/url/something` or`file://base/url/api.raml` will work.

I can:
* Rename argument to `baseUrl`;
* Add comments to examples clarifying that argument purpose.

Reply:
Did this

> 

I see. In that case, let's call it baseUrl instead. Also, here and in the other examples, I would set it to file://base/url/ instead of file://base/url/api.raml to avoid confusion. I say "confusion" because it may lead people to think that we are parsing file://base/url/api.raml when we are not.